core/src/main/java/com/linecorp/armeria/client/HttpRequestSubscriber.java

```diff
@@ -292,10 +232,10 @@ private void write(HttpObject o, boolean endOfStream, boolean flush) {
             state = State.DONE;
         }
 
-        write0(o, endOfStream, flush);
+        write0(o, endOfStream, flush, HttpHeaders.of());
```

Comment:
It seems weird to me that the `write0` is called with `HttpData` and trailers(even though it's empty.)
How about just inlining `write0` method? We can extract the common part to another method then. 🤔

Reply:
I refactored `write0()` and extracted some common to a method. PTAL.